### ISSUE TITLE
<validator>->setName() not applying to AbstractRelated object

### ISSUE THREAD

**Issue by CameronHall** [Respect/Validation#1244]
```php
use Respect\Validation\Validator as v;

$contact = [];
try {
    v::key('firstname', v::notBlank()->setName('First Name'))->check($contact);
} catch (\Exception $e) {
    print_r($e->getMessages());
}
```

Outputs:
```
array(1) {
  [0]=>
  string(30) "Key firstname must be present"
}
```

Expected:
```
array(1) {
  [0]=>
  string(30) "Key First Name must be present"
}
```

**Comment by noobshow** [comment]
I'm betting the "Key" Prefix should not even be there, most intending readers of this validation errors are not the developers themselves, most are users, visitors and clients and as such "should not know a thing is validating a key behind the scene"

**Comment by henriquemoody** [comment]
To all users that were waiting for this fix, thank you very much for your patience! 

You can find this fix on version 2.0.16.

@CameronHall, thank you very much for reporting and also for submitting a pull request! 🐼
